src/hotspot/share/opto/matcher.cpp

```diff
@@ -2794,6 +2794,12 @@ BasicType Matcher::vector_element_basic_type(const MachNode* use, const MachOper
   return def->bottom_type()->is_vect()->element_basic_type();
 }
 
+bool Matcher::is_non_long_integral_vector(const Node* n) {
+  BasicType bt = vector_element_basic_type(n);
+  assert(bt != T_CHAR, "char is not allowed in vector");
+  return is_subword_type(bt) || bt == T_INT;
```

Comment:
I suggest you create a `is_non_long_integral_vector` in `src/hotspot/share/utilities/globalDefinitions.hpp`, next to `is_integral_type`.

Reply:
This method is used only for matcher.  We can do this if  there were more usages of `is_subword_type(bt) || bt == T_INT`.

Reply:
I would do it now. An other person may not see your code and not do the refactoring.